Geometry/HGCalCommonData/src/HGCalDDDConstants.cc

```diff
@@ -1138,34 +1218,140 @@ std::pair<double, double> HGCalDDDConstants::waferPosition(
+      part = (itr->second).part;
+      orient = (itr->second).orient;
+    }
+  } else {
+    auto ktr = hgpar_->typesInLayers_.find(index);
+    if (ktr != hgpar_->typesInLayers_.end())
+      type = hgpar_->waferTypeL_[ktr->second];
+    auto itr = hgpar_->waferTypes_.find(index);
+    if (itr != hgpar_->waferTypes_.end()) {
+      if ((itr->second).second < HGCalWaferMask::k_OffsetRotation) {
+        orient = (itr->second).second;
+        if ((itr->second).first == HGCalGeomTools::k_allCorners) {
+          part = HGCalTypes::WaferFull;
+        } else if ((itr->second).first == HGCalGeomTools::k_fiveCorners) {
+          part = HGCalTypes::WaferFive;
+        } else if ((itr->second).first == HGCalGeomTools::k_fourCorners) {
+          part = HGCalTypes::WaferHalf;
+        } else if ((itr->second).first == HGCalGeomTools::k_threeCorners) {
+          part = HGCalTypes::WaferThree;
+        }
+      } else {
+        part = (itr->second).first;
+        orient = ((itr->second).second - HGCalWaferMask::k_OffsetRotation);
+      }
+    } else {
+      part = HGCalTypes::WaferFull;
+      orient = 0;
+    }
+  }
+  return std::make_tuple(type, part, orient);
+}
+
+std::pair<int, int> HGCalDDDConstants::waferTypeRotation(
+    int layer, int waferU, int waferV, bool fromFile, bool debug) const {
+  int type(HGCalTypes::WaferOut), rotn(0);
+  int wl = HGCalWaferIndex::waferIndex(layer, waferU, waferV);
+  if (fromFile && (waferFileSize() > 0)) {
+    auto itr = hgpar_->waferInfoMap_.find(wl);
+    if (itr != hgpar_->waferInfoMap_.end()) {
+      type = (itr->second).part;
+      rotn = (itr->second).orient;
+    }
+  } else {
+    auto itr = hgpar_->waferTypes_.find(wl);
+    if ((mode_ == HGCalGeometryMode::Hexagon8) || (mode_ == HGCalGeometryMode::Hexagon8Full)) {
+      if (itr != hgpar_->waferTypes_.end()) {
+        if ((itr->second).second < HGCalWaferMask::k_OffsetRotation) {
+          rotn = (itr->second).second;
+          if ((itr->second).first == HGCalGeomTools::k_allCorners) {
+            type = HGCalTypes::WaferFull;
+          } else if ((itr->second).first == HGCalGeomTools::k_fiveCorners) {
+            type = HGCalTypes::WaferFive;
+          } else if ((itr->second).first == HGCalGeomTools::k_fourCorners) {
+            type = HGCalTypes::WaferHalf;
+          } else if ((itr->second).first == HGCalGeomTools::k_threeCorners) {
+            type = HGCalTypes::WaferThree;
+          }
+        } else {
+          type = (itr->second).first;
+          rotn = ((itr->second).second - HGCalWaferMask::k_OffsetRotation);
+        }
+      }
+    }
+  }
+#ifdef EDM_ML_DEBUG
+  if (debug)
+    edm::LogVerbatim("HGCalGeom") << "waferTypeRotation: Layer " << layer << " Wafer " << waferU << ":" << waferV
+                                  << " Index " << std::hex << wl << std::dec << ":"
+                                  << (itr != hgpar_->waferTypes_.end()) << " Type " << type << " Rotation " << rotn;
+#endif
```

Comment:
this broke compilation of static analysis for this package.
Please fix.
Naively, this requires just moving the ifdef in the previous scope (before the ```}``` on L1345

Reply:
Or how about just using `LogTrace`. In the big picture this pattern of hiding non-`LogDebug`-level messages with `#ifdef EDM_ML_DEBUG` is confusing.